nest/src/main.rs

```diff
@@ -129,19 +159,27 @@ fn main() {
         )
         .get_matches();
 
-    if matches.subcommand_matches("pull").is_some() {
-        command::pull::pull(&config);
-    } else if let Some(matches) = matches.subcommand_matches("install") {
-        command::install::install(&config, matches);
-    } else if let Some(matches) = matches.subcommand_matches("uninstall") {
-        command::uninstall::uninstall(matches);
-    } else if let Some(matches) = matches.subcommand_matches("search") {
-        command::search::search(matches);
-    } else if let Some(matches) = matches.subcommand_matches("search") {
-        command::search::search(matches);
-    } else if let Some(matches) = matches.subcommand_matches("upgrade") {
-        command::upgrade::upgrade(matches);
-    } else if let Some(matches) = matches.subcommand_matches("list") {
-        command::list::list(matches);
+    let r = match matches.subcommand() {
+        ("pull", _) => command::pull::pull(&config),
+        ("download", Some(matches)) => command::download::download(&config, matches),
+        ("install", Some(matches)) => command::install::install(&config, matches),
+        _ => unimplemented!(),
+    };
+
+    // All errors arrive here. t's our job to print them on screen and then exit(1).
```

Comment:
Typo, `t's`(probably should be `It's`), while I'm at  it, `on screen` sounds a little superfluous.